packages/vast-xml2js/src/helpers/parseFromString.js

```diff
@@ -0,0 +1,13 @@
+import saneError from 'sane-domparser-error';
+
+const parseFromString = (DOMParser, xmlText) => {
+  const parser = new DOMParser();
```

Comment:
And the instantiate it directly? If it is really required, why not instantiate it before passing into the function? I.e. 
```parseFromString = (parser, text) => ...```

Reply:
parseFromString is forced to know, how to instantiate (if some options needs to be passed), and the only things it needs to know is parseFromString

Reply:
`parseFromString` is there to show that we are indeed doing manual dependency injection.  I just don't agree with you guys. I believe it is important to show that intention in the code.

Reply:
Not sure you guys understand the point of the whole module. which to to have 2 build one for the frontend and one for the backend. I know it may look odd but it shows as best I could that we are doing dependency injection to do `parseFromString` if a developer sees it I expect him to investigate the need for it and eventually find that there are too index files one that uses xmlDom and one that doesn't

Reply:
Tbh, by providing a class object as a dependency injection is wrong in my opinion. In reality, to avoid future complications, and improve code testability, you should provide an object in case. Especially, *performance* wise, you could create a single instance of DOMParser, instead of instantiating it every single call.

Reply:
The performance issue it is a point I agree. Unfortunately on the previous plugin I experience problems trying to reuse the parser therefore I decided against it from the beginning .